operations/security_and_authentication/Change_NCN_Image_Root_Password_and_SSH_Keys_on_PIT_Node.md

```diff
@@ -1,122 +1,197 @@
 
-   If desired, create new SSH keys on the PIT node. These will be copied into the NCN SquashFS images in the next step. Alternatively,
-   copy an existing set of keys and `authorized_hosts` file into a directory for reference in the following step. It is assumed
-   that public keys have a `.pub` extension.
+The new customized images are created in their original image's directory. They have the same name as the original image, except
+with the `secure-` prefix added. The original image is moved into a subdirectory named `old`, for backup purposes.
 
-   Execute the `ncn-image-modification.sh` script located at the top level of the CSM release tarball in order to add SSH keys and
-   set the root password. Optionally, set a local timezone (UTC is the default). If you choose to create new SSH keys, then specify
-   the directory where these keys are located with the `-d` argument to the script, in addition to the other required options.
+There are several choices to be made during this process:
 
-   ```console
-   pit# ${$CSM_PATH}/ncn-image-modification.sh -h
-   Usage: ncn-image-modification.sh [-p] [-d dir] [ -z timezone] [-k kubernetes-squashfs-file] [-s storage-squashfs-file] [ssh-keygen arguments]
+- SSH key files can be provided to the script, or the script can generate them itself.
+- The hashed `root` password can be provided to the script, or the script can prompt for password entry when it is running.
+- To use a non-default timezone, that must be passed into the script.
 
-          This script semi-automates the process of changing the timezone, root
-          password, and adding new SSH keys for the root user to the NCN squashfs
-          image(s).
+### SSH keys
 
-          The script will immediately prompt for a new passphrase for ssh-keygen.
-          The script will then proceed to unsquash the supplied squash files and
-          then prompt for a password. Once the password of the last squash has been
-          provided, the script will continue to completion without interruption.
+#### Script-generated keys
 
-          The process can be fully automated by using the SQUASHFS_ROOT_PW_HASH
-          environment variable (see below) along with either -d or -N.
+To have the script generate the SSH keys automatically, it must be provided with the `ssh-keygen` options to use.
 
-          -a             Do *not* modify the authorized_keys file in the squashfs.
-                         If modifying a previously modified image, or an
-                         authorized_keys file that contains the public key is already
-                         included in the directory used with the -d option, you may
-                         want to use this option.
+- To view the complete list of supported `ssh-keygen` options, view the script usage statement by running it with the `-h` argument.
+- If the `-N` option is not used to specify the passphrase, then the script will prompt for the passphrase when it generates the keys.
```

Comment:
That is true if you don't provide an argument to `-N`... but you may, optionally, provide that argument.  E.g. `-N ''` to have it be blank and not prompt.

Reply:
Yeah, I cover that in one of the examples. Or you did, or whoever wrote the original doc -- I copied that part of that example. But anyway, I can add a sentence here too explaining that any passphrase (including a blank one) is a way to avoid the prompts.

Reply:
Done